ophyd/device.py

```diff
@@ -341,10 +349,16 @@ def describe(self):
 
     def stage(self):
         "Prepare the device to be triggered."
-        if self._staged:
-            raise RuntimeError("Device is already stage. Unstage it first.")
-        self._staged = True
+        if self._staged == Staged.no:
+            pass  # to short-circuit checking individual cases
```

Comment:
First check if there is problem and only then invest the cycles in diagnosing exactly *which* problem we have.